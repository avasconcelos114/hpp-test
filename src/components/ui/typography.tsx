import { ReactNode } from 'react';
import { cn } from '@/lib/utils';

export type TypographySize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
export type TypographyWeight = 'regular' | 'medium' | 'semibold' | 'bold';
export type TypographyTag =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p'
  | 'span'
  | 'label';

interface TypographyProps {
  size: TypographySize;
  tag?: TypographyTag;
  weight?: TypographyWeight;
  children?: ReactNode;
  className?: string;
  htmlFor?: string;
  id?: string;
}

const sizeStyles: Record<TypographySize, string> = {
  xs: 'text-[12px] leading-[16px]',
  sm: 'text-[14px] leading-[22px]',
  md: 'text-[15px] leading-[22px]',
  lg: 'text-[20px] leading-[28px]',
  xl: 'text-[32px] leading-[40px]',
};

const weightStyles: Record<TypographyWeight, string> = {
  regular: 'font-normal',
  medium: 'font-medium',
  semibold: 'font-semibold',
  bold: 'font-bold',
};

export const Typography = ({
  size = 'md',
  tag = 'p',
  weight = 'regular',
  children,
  className = '',
  htmlFor,
  id,
}: TypographyProps) => {
  const Component = tag;
  const sizeClassName = sizeStyles[size];
  const weightClassName = weightStyles[weight];

  return (
    <Component
      className={cn(
        'text-grays-not-quite-black m-0',
        sizeClassName,
        weightClassName,
        className,
      )}
      htmlFor={htmlFor}
      id={id}
    >
      {children}
    </Component>
  );
};
