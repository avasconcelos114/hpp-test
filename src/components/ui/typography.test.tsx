import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Typography } from './typography';

describe('Typography', () => {
  it('should render the typography', () => {
    render(
      <Typography size='sm' weight='regular'>
        Typography
      </Typography>,
    );
    expect(screen.getByText('Typography')).toMatchSnapshot();
  });

  it('should render the typography with the correct size', () => {
    render(
      <Typography size='sm' weight='regular'>
        Typography
      </Typography>,
    );
    const typography = screen.getByText('Typography');
    expect(typography.classList).toContain('text-[14px]');
  });

  it('should render the typography with the correct weight', () => {
    render(
      <Typography size='sm' weight='medium'>
        Typography
      </Typography>,
    );
    const typography = screen.getByText('Typography');
    expect(typography.classList).toContain('font-medium');
  });

  it('should render the typography with the correct color', () => {
    render(
      <Typography size='sm' weight='regular'>
        Typography
      </Typography>,
    );
    const typography = screen.getByText('Typography');
    expect(typography.classList).toContain('text-grays-not-quite-black');
  });

  it('should render the typography with the correct class name', () => {
    render(
      <Typography size='sm' weight='regular' className='text-red-500'>
        Typography
      </Typography>,
    );
    const typography = screen.getByText('Typography');
    expect(typography.classList).toContain('text-red-500');
  });
});
