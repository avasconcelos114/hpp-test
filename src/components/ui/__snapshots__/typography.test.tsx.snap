// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography > should render the typography 1`] = `
<p
  class="text-grays-not-quite-black m-0 text-sm leading-[22px] font-normal"
>
  Typography
</p>
`;
